app/controllers/application_controller.rb

```diff
@@ -1,10 +1,172 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+  	set :session_secret, "secret"
   end
 
-end
\ No newline at end of file
+
+ #Home
+  get '/' do 
+  	erb :'index'
+  end
+
+
+#Sign Up
+  get '/signup' do
+  	if Helpers.is_logged_in?(session)
+  		redirect '/tweets'
+  	else
+  		erb :'users/signup'
+  	end
+  end
+
+  post '/signup' do 
+  	@user = User.create(:username => params[:username], :email => params[:email], :password => params[:password])
+  	@user.save
+  	session[:id] = @user[:id]
+  	if @user.save && !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+  		redirect '/tweets'
+  	else
+  		redirect '/signup'
+  	end
+  end
+
+#Log in
+  get '/login' do
+  	if Helpers.is_logged_in?(session)
+  		redirect '/tweets'
+  	else
+		erb :'users/login'
+	end
+  end
+
+  post '/login' do 
+ 	@user = User.find_by(:username => params[:username])
+ 	if @user && @user.authenticate(params[:password])
+  		session[:id] = @user[:id]
+		redirect '/tweets'
+	else
+		redirect '/login'
+	end
+  end
+
+  #Log Out
+  get '/logout' do 
+  	session.clear
+  	redirect '/login'
+  end
+
+  #Home Page - tweets feed
+  get '/tweets' do
+  	if Helpers.is_logged_in?(session)
+  		@user = User.find_by_id(session[:id])
+  		erb :'tweets/tweets'
+  	else
+  		redirect '/login'
+  	end
+  end
+
+  #Show single user tweets
+  get '/users/:slug' do 
+  	@user = User.find_by_slug(params[:slug])
+  	erb :'users/tweets'
+  end
+
+  #TWEETS
+  #New
+  get '/tweets/new' do 
+  	if Helpers.is_logged_in?(session)
+	  	@user = User.find_by_id(session[:id])
+  		erb :'tweets/new'
+  	else
+  		redirect '/login'
+  	end
+  end
+
+  post '/tweets/new' do 
+  	if params[:content] != ""
+  		@tweet = Tweet.create(:content => params[:content])
+  		@user = User.find_by_id(session[:id])
+  		@user.tweets << @tweet
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```

or alternatively

```
@tweet = current_user.tweets.create(content: params[:content])
```